### ISSUE TITLE
Error: Could not run: (<unknown>): found unexpected end of stream while scanning a quoted scalar at line 57 column 38

### ISSUE THREAD

**Issue by i-am-nut** [Rhizomatica/puppet#5]
After issuing `puppet agent --test` on step 9, basically nothing happens, that's the output:

```
Notice: Starting Puppet client version 3.4.3
Info: Retrieving plugin
Info: Loading facts in /var/lib/puppet/lib/facter/pygpgme.rb
Info: Loading facts in /var/lib/puppet/lib/facter/systemd.rb
Info: Loading facts in /var/lib/puppet/lib/facter/virtualenv_version.rb
Info: Loading facts in /var/lib/puppet/lib/facter/apt_update_last_success.rb
Info: Loading facts in /var/lib/puppet/lib/facter/osreleasemaj.rb
Info: Loading facts in /var/lib/puppet/lib/facter/facter_dot_d.rb
Info: Loading facts in /var/lib/puppet/lib/facter/root_home.rb
Info: Loading facts in /var/lib/puppet/lib/facter/apt_updates.rb
Info: Loading facts in /var/lib/puppet/lib/facter/python_version.rb
Info: Loading facts in /var/lib/puppet/lib/facter/puppet_vardir.rb
Info: Loading facts in /var/lib/puppet/lib/facter/pe_version.rb
Info: Loading facts in /var/lib/puppet/lib/facter/pip_version.rb
Info: Loading facts in /var/lib/puppet/lib/facter/concat_basedir.rb
Info: Caching catalog for 1ed5c1e570c9
Info: Applying configuration version '1502108727'
Notice: Finished catalog run in 0.11 seconds
```

So i tried to apply the .pp file and got the following:

```
root@57913eae361c:/etc/puppet/environments/production/manifests# ls
site.pp
root@57913eae361c:/etc/puppet/environments/production/manifests# puppet apply site.pp
Error: Could not run: (<unknown>): found unexpected end of stream while scanning a quoted scalar at line 57 column 38
root@57913eae361c:/etc/puppet/environments/production/manifests# 
```

I don't know what to do now

**Comment by whyteks** [comment]
Hi. Use puppet version 3.8 please.

**Comment by i-am-nut** [comment]
Hi, i installed Puppet 3.8 but it seems the issues remains
```
root@57913eae361c:/etc/puppet/environments/production/manifests# puppet agent --version 
3.8.7
root@57913eae361c:/etc/puppet/environments/production/manifests# puppet master --version
3.8.7
root@57913eae361c:/etc/puppet/environments/production/manifests# puppet apply --verbose site.pp
Warning: Setting templatedir is deprecated. See http://links.puppetlabs.com/env-settings-deprecations
   (at /usr/lib/ruby/vendor_ruby/puppet/settings.rb:1139:in `issue_deprecation_warning')
Warning: Setting modulepath is deprecated in puppet.conf. See http://links.puppetlabs.com/env-settings-deprecations
   (at /usr/lib/ruby/vendor_ruby/puppet/settings.rb:1141:in `issue_deprecation_warning')
Info: Loading facts
Info: Loading facts
Info: Loading facts
Info: Loading facts
Info: Loading facts
Info: Loading facts
Warning: Could not retrieve fact fqdn
Warning: Host is missing hostname and/or domain: 57913eae361c
Error: Could not run: (<unknown>): found unexpected end of stream while scanning a quoted scalar at line 57 column 38
```


On client, output of `puppet agent --no-daemonize --verbose` is:

```
root@1ed5c1e570c9:/# puppet agent --no-daemonize --verbose
Warning: Setting templatedir is deprecated. See http://links.puppetlabs.com/env-settings-deprecations
   (at /usr/lib/ruby/vendor_ruby/puppet/settings.rb:1139:in `issue_deprecation_warning')
Notice: Starting Puppet client version 3.8.7
Warning: Unable to fetch my node definition, but the agent run will continue:
Warning: getaddrinfo: Name or service not known
Info: Retrieving plugin
Error: /File[/var/lib/puppet/lib]: Failed to generate additional resources using 'eval_generate': getaddrinfo: Name or service not known
Error: /File[/var/lib/puppet/lib]: Could not evaluate: Could not retrieve file metadata for puppet://57913eae361c/plugins: getaddrinfo: Name or service not known
Info: Loading facts
Error: Could not retrieve catalog from remote server: getaddrinfo: Name or service not known
Notice: Using cached catalog
Info: Applying configuration version '1502270803'
Notice: Finished catalog run in 0.15 seconds
Error: Could not send report: getaddrinfo: Name or service not known


```

**Comment by whyteks** [comment]
Hi.
If this is still an issue, please post system information, also output of puppet agent --debug--test and the resultant log from the puppet master.
One could also pay attention to the Warnings.

**Comment by whyteks** [comment]
Maybe read this thread?
https://stackoverflow.com/questions/7780322/puppet-facter-could-not-retrieve-fact-fqdn-how-to-fix-or-circumvent

**Comment by i-am-nut** [comment]
I'm using two Docker containers with Ubuntu 14.04. 
One is the Puppet Master:
- Hostname: 0770d640d0c3

and the other is Puppet Client:
- Hostname: 93f5307619f3


Both are running Puppet 3.8.7.

I just recreated and setup the containers and it seems the problem of `Warning: Could not retrieve fact fqdn` was solved. Output of `facter fqdn` on Puppet Client is `93f5307619f3`, it's own hostname. On Master issuing this command outputs his hostname as well.

While writing this i just solved the issue. 
On `/etc/puppet/hieradata/defaults.yaml` line 57 there was a double quote missing at the end of line:
```
rhizo::advice_email:                 "[ 'your@email.address' ]
```
Putting a double quote at the end of line solved this #5

**But another one arised. Should i create a new issue for this or may we continue it here?** 

The new error is right below:
```
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Evaluation Error: Error while evaluating a Function Call, Could not find data item rhizo::sip_central_ip_address in any Hiera data file and no default supplied at /etc/puppet/environments/production/modules/rhizo_base/manifests/init.pp:66:29 on node 93f5307619f3
```
Here is the erros and warnings of `puppet agent --debug --test`:
```
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Evaluation Error: Error while evaluating a Function Call, Could not find data item rhizo::sip_central_ip_address in any Hiera data file and no default supplied at /etc/puppet/environments/production/modules/rhizo_base/manifests/init.pp:66:29 on node 93f5307619f3
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
```


And here is the output of `puppet master --no-daemonize --verbose` while issuing the previous command on client.
```
root@0770d640d0c3:/etc/puppet/hieradata# puppet master --no-daemonize --verbose
Warning: Setting templatedir is deprecated. See http://links.puppetlabs.com/env-settings-deprecations
   (at /usr/lib/ruby/vendor_ruby/puppet/settings.rb:1139:in `issue_deprecation_warning')
Warning: Setting modulepath is deprecated in puppet.conf. See http://links.puppetlabs.com/env-settings-deprecations
   (at /usr/lib/ruby/vendor_ruby/puppet/settings.rb:1141:in `issue_deprecation_warning')
Notice: Starting Puppet master version 3.8.7



Info: access[^/catalog/([^/]+)$]: allowing 'method' find
Info: access[^/catalog/([^/]+)$]: allowing $1 access
Info: access[^/node/([^/]+)$]: allowing 'method' find
Info: access[^/node/([^/]+)$]: allowing $1 access
Info: access[/certificate_revocation_list/ca]: allowing 'method' find
Info: access[/certificate_revocation_list/ca]: allowing * access
Info: access[^/report/([^/]+)$]: allowing 'method' save
Info: access[^/report/([^/]+)$]: allowing $1 access
Info: access[/file]: allowing * access
Info: access[/certificate/ca]: adding authentication any
Info: access[/certificate/ca]: allowing 'method' find
Info: access[/certificate/ca]: allowing * access
Info: access[/certificate/]: adding authentication any
Info: access[/certificate/]: allowing 'method' find
Info: access[/certificate/]: allowing * access
Info: access[/certificate_request]: adding authentication any
Info: access[/certificate_request]: allowing 'method' find
Info: access[/certificate_request]: allowing 'method' save
Info: access[/certificate_request]: allowing * access
Info: access[/v2.0/environments]: allowing 'method' find
Info: access[/v2.0/environments]: allowing * access
Info: access[/]: adding authentication any
Info: Inserting default '/status' (auth true) ACL
Info: Caching node for 93f5307619f3
Info: Caching node for 93f5307619f3
Error: Evaluation Error: Error while evaluating a Function Call, Could not find data item rhizo::sip_central_ip_address in any Hiera data file and no default supplied at /etc/puppet/environments/production/modules/rhizo_base/manifests/init.pp:66:29 on node 93f5307619f3
Error: Evaluation Error: Error while evaluating a Function Call, Could not find data item rhizo::sip_central_ip_address in any Hiera data file and no default supplied at /etc/puppet/environments/production/modules/rhizo_base/manifests/init.pp:66:29 on node 93f5307619f3
Error: Evaluation Error: Error while evaluating a Function Call, Could not find data item rhizo::sip_central_ip_address in any Hiera data file and no default supplied at /etc/puppet/environments/production/modules/rhizo_base/manifests/init.pp:66:29 on node 93f5307619f3
```

**Comment by whyteks** [comment]
Hi, Apologies, this is an oversight, partly on my part as we don't actually make use of the defaults.yaml in this repo. Please add sip_central_ip_address to your default or node hiera, (and anything else missing that might come up!)

Thanks.
